manifests/key_pair.pp

```diff
@@ -0,0 +1,34 @@
+# @summary Deploy a certificate and its matching key
+# @api private
+define certs::key_pair (
+  Stdlib::Absolutepath $source_dir,
+  Stdlib::Absolutepath $key_file,
+  Stdlib::Absolutepath $cert_file,
+  Enum['present', 'absent'] $key_ensure = 'present',
+  String $key_owner = undef,
+  String $key_group = undef,
+  Stdlib::Filemode $key_mode = '400',
```

Comment:
Here the `$key_mode` defaults to `400` while the `$cert_mode` defaults to `440`

However when this is used under `if $deploy { ... }` in apache.pp, both are set to `0440`. Is there a reason not to use that as the default?

I will check how this is done in the other commits in this PR to better understand.

Reply:
I think my idea had been to default to the most secure mode possible, restricting to read access by the user only. And then force anybody using this to override this default if they need something else. In a lot of our cases, we do `root:service_user` and the `service_user` (e.g. apache) needs to read the key thus making the permissions `440` rather than `400`.

Reply:
The most secure mode is `0440` where the owner is `root` and only the group can read it. This disallows the service itself from overwriting it while still reading it. Remember that if you are the owner of a file, you can change the mode to anything else as well.

This also means you can default the `key_owner` and `cert_owner` to `root` if you always manage them anyway.